src/DynamoCore/Graph/Nodes/NodeModel.cs

```diff
@@ -38,6 +38,25 @@ internal Info(string message, ElementState state = ElementState.Active)
             Message = message;
             State =  state;
         }
+
+        public override bool Equals(object other)
+        {
+            if (other == null) return false;
+
+            var otherInfo = (Info)other;
+            return Message == otherInfo.Message && State == otherInfo.State;
```

Comment:
null checks?

Reply:
why do we need a custom override ?
Isn't the default good enough ? I think the default Equals compares all fields via reflection ...Are you concerned about performance ?

Reply:
I changed it to a class so now we can keep the `Equals` override and perform the `null` checks.